marlowe-playground-client/src/Marlowe/Parser.js

```diff
@@ -10,7 +10,13 @@ exports.parse_ = function (emptyInputError, parserError, success, fs, input) {
     const parser = new nearley.Parser(nearley.Grammar.fromCompiled(grammar(fs)));
     try {
         parser.feed(input);
-        return success(parser.results[0]);
+        // When the top level contract is valid but incomplete, the parser.results is an empty array,
+        // which caused a runtime error.
+        if (parser.results.length == 0) {
+          return parserError({ message: "Incomplete top level contract", row: 1, column: 0, token: "" });
+        } else {
+          return success(parser.results[0]);
+        }
```

Comment:
While you're here you could tidy the spacing in the `getErrorMessage` function below... But maybe not worth waiting for hydra just for that. 🤷‍♂️